From e46cfb98b6513a040387a93bfb8269a359d2bc76 Mon Sep 17 00:00:00 2001
From: QbProg <tholag@gmail.com>
Date: Sun, 24 Jul 2011 17:54:00 +0200
Subject: [PATCH 01/11] [warning-fix] MSVC pragmas

Removed many MSVC warning pragmas, that previously hide some possible warning fixes.
Corrected some others to be disabled instead of level 4 (actually used to do the fixes)
---
 drv/ShapeSchema/ShapeSchema.cxx                    |    5 +++++
 drv/StdLSchema/StdLSchema.cxx                      |    5 +++++
 drv/StdSchema/StdSchema.cxx                        |    5 +++++
 drv/XCAFSchema/XCAFSchema.cxx                      |    5 +++++
 src/BOPTools/BOPTools_PaveSet.cxx                  |    7 +------
 src/BinDrivers/BinDrivers.cxx                      |    2 +-
 src/BinLDrivers/BinLDrivers.cxx                    |    2 +-
 src/BinTObjDrivers/BinTObjDrivers.cxx              |    2 +-
 src/BinXCAFDrivers/BinXCAFDrivers.cxx              |    2 +-
 .../BooleanOperations_ShapesDataStructure.cxx      |   10 ----------
 src/FWOSDriver/FWOSDriver.cxx                      |    2 +-
 src/IntTools/IntTools.cxx                          |    7 -------
 src/IntTools/IntTools_EdgeFace.cxx                 |    8 --------
 src/LDOM/LDOM_BasicAttribute.cxx                   |    5 -----
 src/LDOM/LDOM_BasicElement.cxx                     |    5 -----
 src/LDOM/LDOM_BasicText.cxx                        |    5 -----
 src/OSD/OSD_Environment.cxx                        |    4 ----
 src/OSD/OSD_Process.cxx                            |    4 ----
 src/PlotMgt/PlotMgt_TextManager.cxx                |    4 ----
 src/Poly/Poly_CoherentNode.cxx                     |    4 ----
 src/Poly/Poly_CoherentTriPtr.cxx                   |    4 ----
 src/StdDrivers/StdDrivers.cxx                      |    2 +-
 src/StdLDrivers/StdLDrivers.cxx                    |    2 +-
 src/TDF/TDF_Data.cxx                               |    9 ---------
 src/TDF/TDF_Label.cxx                              |    9 ---------
 src/TopExp/TopExp_Explorer.cxx                     |    8 --------
 src/ViewerTest/ViewerTest.cxx                      |    1 -
 src/ViewerTest/ViewerTest_ObjectCommands.cxx       |    1 -
 src/ViewerTest/ViewerTest_ViewerCommands.cxx       |    1 -
 src/VrmlData/VrmlData_Geometry.cxx                 |    1 -
 src/VrmlData/VrmlData_Group.cxx                    |    1 -
 src/VrmlData/VrmlData_IndexedFaceSet.cxx           |    1 -
 src/VrmlData/VrmlData_IndexedLineSet.cxx           |    1 -
 src/VrmlData/VrmlData_Material.cxx                 |    1 -
 src/VrmlData/VrmlData_Node.cxx                     |    1 -
 src/VrmlData/VrmlData_Scene.cxx                    |    1 -
 src/VrmlData/VrmlData_WorldInfo.cxx                |    1 -
 src/XCAFDrivers/XCAFDrivers.cxx                    |    2 +-
 src/XmlDrivers/XmlDrivers.cxx                      |    2 +-
 src/XmlLDrivers/XmlLDrivers.cxx                    |    2 +-
 src/XmlTObjDrivers/XmlTObjDrivers.cxx              |    2 +-
 src/XmlXCAFDrivers/XmlXCAFDrivers.cxx              |    2 +-
 42 files changed, 33 insertions(+), 115 deletions(-)

From 1bfed73424019d8e17502623c94ea06bac4cbf7a Mon Sep 17 00:00:00 2001
From: salajkav-dlubalcz <vojtech.salajka@dlubal.cz>
Date: Wed, 12 Sep 2012 09:48:31 +0200
Subject: [PATCH 1/4] MinGW32 compatibility fixes.

---
 inc/Standard_Atomic.hxx |    4 ++--
 src/OSD/OSD_MemInfo.cxx |    4 ++++
 2 files changed, 6 insertions(+), 2 deletions(-)

From 119d575b5b78b7b57196755cd50ae581ef03a191 Mon Sep 17 00:00:00 2001
From: salajkav-dlubalcz <vojtech.salajka@dlubal.cz>
Date: Wed, 12 Sep 2012 11:22:32 +0200
Subject: [PATCH 2/4] Source code simplification.

---
 inc/Standard_Atomic.hxx |   37 +++++++++++++++++--------------------
 1 file changed, 17 insertions(+), 20 deletions(-)

From 6eb52d540227e3fddfcab5009306adb3399c9442 Mon Sep 17 00:00:00 2001
From: salajkav-dlubalcz <vojtech.salajka@dlubal.cz>
Date: Wed, 12 Sep 2012 11:26:38 +0200
Subject: [PATCH 3/4] Tab/Space fix.

---
 inc/Standard_Atomic.hxx |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

From 907cea25e7b538e9ee1fc084f3e25bbab032b619 Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Fri, 28 Sep 2012 09:00:01 +0200
Subject: [PATCH 4/4] Cosmetics: rearrange Standard_Atomic.hxx to minimize
 diff against OCCT

Remove trailing whitespace in OSD_MemInfo.cxx.
---
 inc/Standard_Atomic.hxx |    9 ++++++---
 src/OSD/OSD_MemInfo.cxx |    4 ++--
 2 files changed, 8 insertions(+), 5 deletions(-)

Index: oce/drv/ExprIntrp/ExprIntrp.tab.c
===================================================================
--- oce.orig/drv/ExprIntrp/ExprIntrp.tab.c
+++ oce/drv/ExprIntrp/ExprIntrp.tab.c
@@ -22,6 +22,11 @@
  by  GNU Bison version 1.25
   */
 
+/* Disable warnings in this auto-generated file */
+#ifdef _MSC_VER
+#pragma warning(push,0)
+#endif
+
 #define YYBISON 1  /* Identify Bison output.  */
 
 #define yyparse ExprIntrpparse
@@ -1248,3 +1253,7 @@ yyerrhandle:
   goto yynewstate;
 }
 #line 150 "j:/kas/c40/ros/src/exprintrp/exprintrp.yacc"
+
+#ifdef _MSC_VER
+#pragma warning(pop)
+#endif
Index: oce/drv/ExprIntrp/lex.ExprIntrp.c
===================================================================
--- oce.orig/drv/ExprIntrp/lex.ExprIntrp.c
+++ oce/drv/ExprIntrp/lex.ExprIntrp.c
@@ -1,3 +1,8 @@
+/* Disable warnings in this auto-generated file */
+#ifdef _MSC_VER
+#pragma warning(push,0)
+#endif
+
 #define yy_create_buffer ExprIntrp_create_buffer
 #define yy_delete_buffer ExprIntrp_delete_buffer
 #define yy_scan_buffer ExprIntrp_scan_buffer
@@ -3230,3 +3235,7 @@ int main()
 	return 0;
 	}
 #endif
+
+#ifdef _MSC_VER
+#pragma warning(pop)
+#endif
Index: oce/drv/ShapeSchema/ShapeSchema.cxx
===================================================================
--- oce.orig/drv/ShapeSchema/ShapeSchema.cxx
+++ oce/drv/ShapeSchema/ShapeSchema.cxx
@@ -518,6 +518,11 @@
 #include <ShapeSchema_PCollection_HExtendedString.hxx>
 #endif
 
+// avoid warnings on 'extern "C"' functions returning C++ classes
+#ifdef _MSC_VER
+#pragma warning(disable:4190)
+#endif
+
 IMPLEMENT_STANDARD_HANDLE(ShapeSchema,Storage_Schema)
 IMPLEMENT_STANDARD_RTTIEXT(ShapeSchema,Storage_Schema)
 
Index: oce/drv/StdLSchema/StdLSchema.cxx
===================================================================
--- oce.orig/drv/StdLSchema/StdLSchema.cxx
+++ oce/drv/StdLSchema/StdLSchema.cxx
@@ -161,6 +161,11 @@
 #include <StdLSchema_PColStd_HArray2OfInteger.hxx>
 #endif
 
+// avoid warnings on 'extern "C"' functions returning C++ classes
+#ifdef _MSC_VER
+#pragma warning(disable:4190)
+#endif
+
 IMPLEMENT_STANDARD_HANDLE(StdLSchema,Storage_Schema)
 IMPLEMENT_STANDARD_RTTIEXT(StdLSchema,Storage_Schema)
 
Index: oce/drv/StdSchema/StdSchema.cxx
===================================================================
--- oce.orig/drv/StdSchema/StdSchema.cxx
+++ oce/drv/StdSchema/StdSchema.cxx
@@ -95,6 +95,11 @@
 #include <StdSchema_PTopLoc_ItemLocation.hxx>
 #endif
 
+// avoid warnings on 'extern "C"' functions returning C++ classes
+#ifdef _MSC_VER
+#pragma warning(disable:4190)
+#endif
+
 IMPLEMENT_STANDARD_HANDLE(StdSchema,Storage_Schema)
 IMPLEMENT_STANDARD_RTTIEXT(StdSchema,Storage_Schema)
 
Index: oce/drv/StepFile/lex.step.c
===================================================================
--- oce.orig/drv/StepFile/lex.step.c
+++ oce/drv/StepFile/lex.step.c
@@ -1,3 +1,8 @@
+/* Disable warnings in this auto-generated file */
+#ifdef _MSC_VER
+#pragma warning(push,0)
+#endif
+
 #define yy_create_buffer step_create_buffer
 #define yy_delete_buffer step_delete_buffer
 #define yy_scan_buffer step_scan_buffer
@@ -1748,3 +1753,7 @@ int main()
 	return 0;
 	}
 #endif
+
+#ifdef _MSC_VER
+#pragma warning(pop)
+#endif
Index: oce/drv/StepFile/step.tab.c
===================================================================
--- oce.orig/drv/StepFile/step.tab.c
+++ oce/drv/StepFile/step.tab.c
@@ -34,6 +34,12 @@
 
 #include "recfile.ph"		/* definitions des types d'arguments */
 #include "recfile.pc"		/* la-dedans, tout y est */
+
+/* Disable warnings in this auto-generated file */
+#ifdef _MSC_VER
+#pragma warning(push,0)
+#endif
+
 /*
 #define stepparse STEPparse
 #define steplex STEPlex
@@ -957,3 +963,8 @@ yyerrhandle:
   goto yynewstate;
 }
 #line 146 "/misc/dn60/WOKLOC/wok_entities/LOC/dev/CR000655-CR655-products-occt/src/StepFile/step.yacc"
+
+
+#ifdef _MSC_VER
+#pragma warning(pop)
+#endif
Index: oce/drv/XCAFSchema/XCAFSchema.cxx
===================================================================
--- oce.orig/drv/XCAFSchema/XCAFSchema.cxx
+++ oce/drv/XCAFSchema/XCAFSchema.cxx
@@ -83,6 +83,11 @@
 #include <XCAFSchema_PCollection_HExtendedString.hxx>
 #endif
 
+// avoid warnings on 'extern "C"' functions returning C++ classes
+#ifdef _MSC_VER
+#pragma warning(disable:4190)
+#endif
+
 IMPLEMENT_STANDARD_HANDLE(XCAFSchema,Storage_Schema)
 IMPLEMENT_STANDARD_RTTIEXT(XCAFSchema,Storage_Schema)
 
Index: oce/inc/Standard_Atomic.hxx
===================================================================
--- oce.orig/inc/Standard_Atomic.hxx
+++ oce/inc/Standard_Atomic.hxx
@@ -34,20 +34,20 @@
 
 #include <Standard_Macro.hxx>
 
-#if defined(_MSC_VER) || defined(__BORLANDC__)
-#ifdef __BORLANDC__
-extern "C" {
-  __declspec(dllimport) long __stdcall InterlockedIncrement ( long volatile *lpAddend);
-  __declspec(dllimport) long __stdcall InterlockedDecrement ( long volatile *lpAddend);
- }
- #define _InterlockedIncrement InterlockedIncrement
- #define _InterlockedDecrement InterlockedDecrement
-#elif defined(_MSC_VER)
- extern "C" {
-  long _InterlockedIncrement(long volatile* lpAddend);
-  long _InterlockedDecrement(long volatile* lpAddend);
- }
-#endif
+#if (defined(_WIN32) || defined(__WIN32__))
+  #ifdef _MSC_VER
+    extern "C" {
+      long _InterlockedIncrement(long volatile* lpAddend);
+      long _InterlockedDecrement(long volatile* lpAddend);
+    }
+  #else
+    extern "C" {
+      __declspec(dllimport) long __stdcall InterlockedIncrement ( long volatile *lpAddend);
+      __declspec(dllimport) long __stdcall InterlockedDecrement ( long volatile *lpAddend);
+    }
+    #define _InterlockedIncrement InterlockedIncrement
+    #define _InterlockedDecrement InterlockedDecrement
+  #endif
 #endif
 
 #if defined(_MSC_VER)
Index: oce/src/BOPTools/BOPTools_PaveSet.cxx
===================================================================
--- oce.orig/src/BOPTools/BOPTools_PaveSet.cxx
+++ oce/src/BOPTools/BOPTools_PaveSet.cxx
@@ -26,9 +26,6 @@
 #include <BOPTools_QuickSortPave.hxx>
 #include <BOPTools_ComparePave.hxx>
 
-#ifdef _MSC_VER
-#pragma warning ( disable : 4101 )
-#endif
 
 //=======================================================================
 // function: BOPTools_PaveSet::BOPTools_PaveSet
@@ -89,6 +86,4 @@
     }
   }
 }
-#ifdef _MSC_VER
-#pragma warning ( default : 4101 )
-#endif
+
Index: oce/src/BinDrivers/BinDrivers.cxx
===================================================================
--- oce.orig/src/BinDrivers/BinDrivers.cxx
+++ oce/src/BinDrivers/BinDrivers.cxx
@@ -34,7 +34,7 @@
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
 #ifdef _MSC_VER
-#pragma warning(4:4190)
+#pragma warning(disable:4190)
 #endif
 
 static Standard_GUID BinStorageDriver  ("03a56835-8269-11d5-aab2-0050044b1af1");
Index: oce/src/BinLDrivers/BinLDrivers.cxx
===================================================================
--- oce.orig/src/BinLDrivers/BinLDrivers.cxx
+++ oce/src/BinLDrivers/BinLDrivers.cxx
@@ -31,7 +31,7 @@
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
 #ifdef _MSC_VER
-#pragma warning(4:4190)
+#pragma warning(disable:4190)
 #endif
 
 static Standard_GUID BinLStorageDriver  ("13a56835-8269-11d5-aab2-0050044b1af1");
Index: oce/src/BinTObjDrivers/BinTObjDrivers.cxx
===================================================================
--- oce.orig/src/BinTObjDrivers/BinTObjDrivers.cxx
+++ oce/src/BinTObjDrivers/BinTObjDrivers.cxx
@@ -33,7 +33,7 @@
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
 #ifdef _MSC_VER
-#pragma warning(4:4190)
+#pragma warning(disable:4190)
 #endif
 
 static Standard_GUID BinStorageDriver  ("f78ff4a2-a779-11d5-aab4-0050044b1af1");
Index: oce/src/BinXCAFDrivers/BinXCAFDrivers.cxx
===================================================================
--- oce.orig/src/BinXCAFDrivers/BinXCAFDrivers.cxx
+++ oce/src/BinXCAFDrivers/BinXCAFDrivers.cxx
@@ -29,7 +29,7 @@
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
 #ifdef _MSC_VER
-#pragma warning(4:4190)
+#pragma warning(disable:4190)
 #endif
 
 static Standard_GUID BinXCAFStorageDriver  ("a78ff496-a779-11d5-aab4-0050044b1af1");
Index: oce/src/BooleanOperations/BooleanOperations_ShapesDataStructure.cxx
===================================================================
--- oce.orig/src/BooleanOperations/BooleanOperations_ShapesDataStructure.cxx
+++ oce/src/BooleanOperations/BooleanOperations_ShapesDataStructure.cxx
@@ -58,11 +58,6 @@
 #include <TColStd_MapOfInteger.hxx>
 #include <BOPTColStd_Dump.hxx>
 
-
-#ifdef _MSC_VER
-#pragma warning ( disable : 4291 )
-#endif
-
 static
   void Message(const Standard_Integer i);
 
@@ -902,11 +897,6 @@ void Message(const Standard_Integer i)
 }
 //modified by NIZNHY-PKV Wed Feb  2 14:44:08 2005f
 /*
-#ifdef WNT
-#pragma warning ( default : 4291 )
-#endif
-*/
-/*
 //===========================================================================
 //function : InsertInterference
 //purpose  : 
Index: oce/src/FWOSDriver/FWOSDriver.cxx
===================================================================
--- oce.orig/src/FWOSDriver/FWOSDriver.cxx
+++ oce/src/FWOSDriver/FWOSDriver.cxx
@@ -27,7 +27,7 @@
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
 #ifdef _MSC_VER
-#pragma warning(4:4190)
+#pragma warning(disable:4190)
 #endif
 
 PLUGIN(FWOSDriver)
Index: oce/src/IntTools/IntTools.cxx
===================================================================
--- oce.orig/src/IntTools/IntTools.cxx
+++ oce/src/IntTools/IntTools.cxx
@@ -38,9 +38,6 @@
 #include <TColStd_ListIteratorOfListOfReal.hxx>
 #include <gce_ErrorType.hxx>
 
-#ifdef _MSC_VER
-#pragma warning ( disable : 4101 )
-#endif
 //=======================================================================
 //function : IntTools::GetRadius
 //purpose  : 
@@ -377,7 +374,3 @@
   }
   return 1;
 }
-
-#ifdef _MSC_VER
-#pragma warning ( default : 4101 )
-#endif
Index: oce/src/IntTools/IntTools_EdgeFace.cxx
===================================================================
--- oce.orig/src/IntTools/IntTools_EdgeFace.cxx
+++ oce/src/IntTools/IntTools_EdgeFace.cxx
@@ -69,10 +69,6 @@
 #include <GeomAdaptor_HSurface.hxx>
 #include <IntCurveSurface_IntersectionPoint.hxx>
 
-#ifdef _MSC_VER
-#pragma warning ( disable : 4101 )
-#endif
-
 static
   Standard_Boolean IsCoplanar (const BRepAdaptor_Curve&  ,
 			       const BRepAdaptor_Surface& );
@@ -1566,7 +1562,3 @@ Standard_Integer AdaptiveDiscret (const
   return iDiscretNew;
 }
 
-
-#ifdef _MSC_VER
-#pragma warning ( default : 4101 )
-#endif
Index: oce/src/OSD/OSD_Environment.cxx
===================================================================
--- oce.orig/src/OSD/OSD_Environment.cxx
+++ oce/src/OSD/OSD_Environment.cxx
@@ -245,10 +245,6 @@ Standard_Integer OSD_Environment::Error(
 
 #include <stdio.h>
 
-#ifdef _MSC_VER
-#pragma warning( disable : 4700 )
-#endif
-
 static void __fastcall _set_error ( OSD_Error&, DWORD );
 
 OSD_Environment :: OSD_Environment () {
Index: oce/src/OSD/OSD_MemInfo.cxx
===================================================================
--- oce.orig/src/OSD/OSD_MemInfo.cxx
+++ oce/src/OSD/OSD_MemInfo.cxx
@@ -18,6 +18,10 @@
 // and conditions governing the rights and limitations under the License.
 
 #if (defined(_WIN32) || defined(__WIN32__))
+  #if defined(__MINGW32__)
+    #define WIN32_WINNT 0x0500
+    #define WINVER 0x0500
+  #endif
   #include <windows.h>
   #include <winbase.h>
   #include <process.h>
Index: oce/src/OSD/OSD_Process.cxx
===================================================================
--- oce.orig/src/OSD/OSD_Process.cxx
+++ oce/src/OSD/OSD_Process.cxx
@@ -239,10 +239,6 @@ Standard_Integer OSD_Process::Error()con
 #include <stdint.h>
 #endif
 
-#ifdef _MSC_VER
-#pragma warning( disable : 4700 )
-#endif
-
 void _osd_wnt_set_error ( OSD_Error&, OSD_WhoAmI, ... );
 
 OSD_Process :: OSD_Process () {
Index: oce/src/PlotMgt/PlotMgt_TextManager.cxx
===================================================================
--- oce.orig/src/PlotMgt/PlotMgt_TextManager.cxx
+++ oce/src/PlotMgt/PlotMgt_TextManager.cxx
@@ -21,10 +21,6 @@
 #include <PlotMgt_TextManager.ixx>
 #include <Aspect_Units.hxx>
 
-#ifdef _MSC_VER
-#pragma warning (disable : 4244)
-#endif
-
 //#define TRACE
 #define MAXPATHPOINTS  800
 #define MAXCHARPATHS   20
Index: oce/src/Poly/Poly_CoherentNode.cxx
===================================================================
--- oce.orig/src/Poly/Poly_CoherentNode.cxx
+++ oce/src/Poly/Poly_CoherentNode.cxx
@@ -22,10 +22,6 @@
 #include <Poly_CoherentNode.hxx>
 #include <Poly_CoherentTriangle.hxx>
 
-#ifdef _MSC_VER
-#pragma warning(disable:4996)
-#endif
-
 #include <stdio.h>
 
 //=======================================================================
Index: oce/src/StdDrivers/StdDrivers.cxx
===================================================================
--- oce.orig/src/StdDrivers/StdDrivers.cxx
+++ oce/src/StdDrivers/StdDrivers.cxx
@@ -32,7 +32,7 @@
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
 #ifdef _MSC_VER
-#pragma warning(4:4190)
+#pragma warning(disable:4190)
 #endif
 
 static Standard_GUID StdStorageDriver     ("ad696000-5b34-11d1-b5ba-00a0c9064368");
Index: oce/src/StdLDrivers/StdLDrivers.cxx
===================================================================
--- oce.orig/src/StdLDrivers/StdLDrivers.cxx
+++ oce/src/StdLDrivers/StdLDrivers.cxx
@@ -31,7 +31,7 @@
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
 #ifdef _MSC_VER
-#pragma warning(4:4190)
+#pragma warning(disable:4190)
 #endif
 
 static Standard_GUID StdLStorageDriver     ("bd696000-5b34-11d1-b5ba-00a0c9064368");
Index: oce/src/TopExp/TopExp_Explorer.cxx
===================================================================
--- oce.orig/src/TopExp/TopExp_Explorer.cxx
+++ oce/src/TopExp/TopExp_Explorer.cxx
@@ -143,11 +143,6 @@ const TopoDS_Shape&  TopExp_Explorer::Cu
 //purpose  : 
 //=======================================================================
 
-#ifdef _MSC_VER
-#pragma warning(push)
-#pragma warning(disable: 4291) // to avoid warning when using new(buffer) syntax
-#endif
-
 void  TopExp_Explorer::Next()
 {
   Standard_Integer NewSize;
@@ -225,9 +220,6 @@ void  TopExp_Explorer::Next()
   hasMore = Standard_False;
 }
 
-#ifdef _MSC_VER
-#pragma warning(pop)
-#endif
 
 //=======================================================================
 //function : ReInit
Index: oce/src/ViewerTest/ViewerTest.cxx
===================================================================
--- oce.orig/src/ViewerTest/ViewerTest.cxx
+++ oce/src/ViewerTest/ViewerTest.cxx
@@ -75,7 +75,6 @@
 #ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
 #pragma warning(4:4190)
-#pragma warning (disable:4996)
 #endif
 
 #include <NIS_InteractiveContext.hxx>
Index: oce/src/ViewerTest/ViewerTest_ObjectCommands.cxx
===================================================================
--- oce.orig/src/ViewerTest/ViewerTest_ObjectCommands.cxx
+++ oce/src/ViewerTest/ViewerTest_ObjectCommands.cxx
@@ -146,7 +146,6 @@
 
 #ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
-#pragma warning (disable:4996)
 #endif
 
 extern ViewerTest_DoubleMapOfInteractiveAndName& GetMapOfAIS();
Index: oce/src/ViewerTest/ViewerTest_ViewerCommands.cxx
===================================================================
--- oce.orig/src/ViewerTest/ViewerTest_ViewerCommands.cxx
+++ oce/src/ViewerTest/ViewerTest_ViewerCommands.cxx
@@ -78,7 +78,6 @@
 
 #ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
-#pragma warning (disable:4996)
 #endif
 
 #endif
Index: oce/src/VrmlData/VrmlData_Geometry.cxx
===================================================================
--- oce.orig/src/VrmlData/VrmlData_Geometry.cxx
+++ oce/src/VrmlData/VrmlData_Geometry.cxx
@@ -40,7 +40,6 @@
 
 #ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
-#pragma warning (disable:4996)
 #endif
 
 #include <stdio.h>
Index: oce/src/VrmlData/VrmlData_Group.cxx
===================================================================
--- oce.orig/src/VrmlData/VrmlData_Group.cxx
+++ oce/src/VrmlData/VrmlData_Group.cxx
@@ -31,7 +31,6 @@
 
 #ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
-#pragma warning (disable:4996)
 #endif
 
 #include <stdio.h>
Index: oce/src/VrmlData/VrmlData_IndexedFaceSet.cxx
===================================================================
--- oce.orig/src/VrmlData/VrmlData_IndexedFaceSet.cxx
+++ oce/src/VrmlData/VrmlData_IndexedFaceSet.cxx
@@ -37,7 +37,6 @@
 
 #ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
-#pragma warning (disable:4996)
 #endif
 
 #include <stdio.h>
Index: oce/src/VrmlData/VrmlData_IndexedLineSet.cxx
===================================================================
--- oce.orig/src/VrmlData/VrmlData_IndexedLineSet.cxx
+++ oce/src/VrmlData/VrmlData_IndexedLineSet.cxx
@@ -32,7 +32,6 @@
 
 #ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
-#pragma warning (disable:4996)
 #endif
 
 IMPLEMENT_STANDARD_HANDLE  (VrmlData_IndexedLineSet, VrmlData_Geometry)
Index: oce/src/VrmlData/VrmlData_Material.cxx
===================================================================
--- oce.orig/src/VrmlData/VrmlData_Material.cxx
+++ oce/src/VrmlData/VrmlData_Material.cxx
@@ -26,7 +26,6 @@
 
 #ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
-#pragma warning (disable:4996)
 #endif
 
 #include <stdio.h>
Index: oce/src/VrmlData/VrmlData_Node.cxx
===================================================================
--- oce.orig/src/VrmlData/VrmlData_Node.cxx
+++ oce/src/VrmlData/VrmlData_Node.cxx
@@ -31,7 +31,6 @@
 
 #ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
-#pragma warning (disable:4996)
 #endif
 
 IMPLEMENT_STANDARD_HANDLE  (VrmlData_Node, Standard_Transient)
Index: oce/src/VrmlData/VrmlData_Scene.cxx
===================================================================
--- oce.orig/src/VrmlData/VrmlData_Scene.cxx
+++ oce/src/VrmlData/VrmlData_Scene.cxx
@@ -50,7 +50,6 @@
 
 #ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
-#pragma warning (disable:4996)
 #endif
 
 #include <stdio.h>
Index: oce/src/VrmlData/VrmlData_WorldInfo.cxx
===================================================================
--- oce.orig/src/VrmlData/VrmlData_WorldInfo.cxx
+++ oce/src/VrmlData/VrmlData_WorldInfo.cxx
@@ -25,7 +25,6 @@
 
 #ifdef _MSC_VER
 #define _CRT_SECURE_NO_DEPRECATE
-#pragma warning (disable:4996)
 #endif
 
 #include <stdio.h>
Index: oce/src/XCAFDrivers/XCAFDrivers.cxx
===================================================================
--- oce.orig/src/XCAFDrivers/XCAFDrivers.cxx
+++ oce/src/XCAFDrivers/XCAFDrivers.cxx
@@ -36,7 +36,7 @@
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
 #ifdef _MSC_VER
-#pragma warning(4:4190)
+#pragma warning(disable:4190)
 #endif
 
 static Standard_GUID XSStorageDriver  ("ed8793f8-3142-11d4-b9b5-0060b0ee281b");
Index: oce/src/XmlDrivers/XmlDrivers.cxx
===================================================================
--- oce.orig/src/XmlDrivers/XmlDrivers.cxx
+++ oce/src/XmlDrivers/XmlDrivers.cxx
@@ -40,7 +40,7 @@
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
 #ifdef _MSC_VER
-#pragma warning(4:4190)
+#pragma warning(disable:4190)
 #endif
 
 static Standard_GUID XmlStorageDriver  ("03a56820-8269-11d5-aab2-0050044b1af1");
Index: oce/src/XmlLDrivers/XmlLDrivers.cxx
===================================================================
--- oce.orig/src/XmlLDrivers/XmlLDrivers.cxx
+++ oce/src/XmlLDrivers/XmlLDrivers.cxx
@@ -35,7 +35,7 @@
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
 #ifdef _MSC_VER
-#pragma warning(4:4190)
+#pragma warning(disable:4190)
 #endif
 
 static Standard_GUID XmlLStorageDriver  ("13a56820-8269-11d5-aab2-0050044b1af1");
Index: oce/src/XmlTObjDrivers/XmlTObjDrivers.cxx
===================================================================
--- oce.orig/src/XmlTObjDrivers/XmlTObjDrivers.cxx
+++ oce/src/XmlTObjDrivers/XmlTObjDrivers.cxx
@@ -34,7 +34,7 @@
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
 #ifdef _MSC_VER
-#pragma warning(4:4190)
+#pragma warning(disable:4190)
 #endif
 
 static Standard_GUID XmlStorageDriver  ("f78ff4a0-a779-11d5-aab4-0050044b1af1");
Index: oce/src/XmlXCAFDrivers/XmlXCAFDrivers.cxx
===================================================================
--- oce.orig/src/XmlXCAFDrivers/XmlXCAFDrivers.cxx
+++ oce/src/XmlXCAFDrivers/XmlXCAFDrivers.cxx
@@ -29,7 +29,7 @@
 
 // avoid warnings on 'extern "C"' functions returning C++ classes
 #ifdef _MSC_VER
-#pragma warning(4:4190)
+#pragma warning(disable:4190)
 #endif
 
 static Standard_GUID XSStorageDriver  ("f78ff496-a779-11d5-aab4-0050044b1af1");
